nixos/modules/services/misc/klipper.nix

```diff
@@ -35,6 +35,38 @@ in
         description = lib.mdDoc "Path of the API socket to create.";
       };
 
+      isConfigMutable = mkOption {
+        type = types.bool;
+        default = false;
+        example = true;
+        description = lib.mdDoc ''
+          Whether to copy config to mutable folder instead of using one directly from the store.
+          This will only copy config from NixOS configuration if file at `services.klipper.mutableConfigPath` doesn't exist.
+        '';
+      };
+
+      mutableConfigPath = mkOption {
+        type = types.path;
+        default = "/var/lib/klipper/printer.cfg";
+        description = lib.mdDoc "Path to mutable Klipper config file.";
+      };
+
+      configFile = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = lib.mdDoc ''
+          Path to default Klipper config.
+        '';
+      };
```

Comment:
Yes — people might want to use some other folder. Config file is used as a source file — one that gets copied to `mutableConfigPath` if config is mutable, or to `/etc/klipper.cfg` if it is immutable.

Reply:
3 options are quite excessive for this functionality. We can easily condense this into two or one if we wanted.

Reply:
Well, we can use `mutableConfigFolder` instead of gcode/config paths and just use default names in it always. 
But I didn't see any reason to do so.

And keep in mind that this PR has two separable functionalities really — one is an ability to specify `configFile` instead of `settings`; the other one is a way to store config in a mutable directory. 
I can technically make two separate PRs for those, but still.